packages/gatsby-react-router-scroll/src/StateStorage.js

```diff
@@ -46,8 +46,9 @@ export default class SessionStorage {
   }
 
   getStateKey(location, key) {
-    const locationKey = location.key
-    const stateKeyBase = `${STATE_KEY_PREFIX}${locationKey}`
+    const locationPath = location.pathname
+    const strippedLocationPath = locationPath.endsWith(`/`) ? locationPath.slice(0, -1) : locationPath
```

Comment:
`endsWith` isn't on IE https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/endsWith so just do slice(-1) or whatever it is exactly